### DISCUSSION TITLE
topic visualized

### DISCUSSION THREAD

**Discussion by qinian9584** [foxglove/community#59] [Q&A]
Where can I transform messages that cannot be visualized and forward them.

Similar to node playground

**Comment by defunctzombie** [comment]
Please elaborate on your feature request. I do not understand what you are asking.

**Comment by qinian9584** [comment]
> 
I'm developing foxglove


Where to process the message so that the message can be visualized in 3D penel


That is, how to convert the data into a marker and issue it

**Reply by jtbandes** [reply]
For the current 3D panel, see https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/ThreeDimensionalViz/SceneBuilder/index.ts. For the new 3D panel, see https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/ThreeDeeRender/Renderer.ts#L248.
